Docs/source/running_cpp/parameters.rst

```diff
@@ -974,6 +974,26 @@ Numerics and algorithms
 
      If ``algo.maxwell_fdtd_solver`` is not specified, ``yee`` is the default.
 
+* ``algo.em_solver_medium`` (`string`, optional)
+    The medium for evaluating the Maxwell solver. Available options are :
+
+    - ``vacuum``: vacuum properties are used in the Maxwell solver.
+    - ``macroscopic``: macroscopic Maxwell equation is evaluated. If this option is selected, then the corresponding properties of the medium must be provided using ``macroscopic.sigma``, ``macroscopic.epsilon``, and ``macroscopic.mu``.
+
+    If ``algo.em_solver_medium`` is not specified, ``vacuum`` is the default.
+
+* ``algo.macroscopic_sigma_method`` (`string`, optional)
+    The algorithm for updating electric field when ``algo.em_solver_medium`` is macroscopic. Available options are:
+
+    - ``laxwendroff`` is the semi-implicit scheme for E-update.
```

Comment:
Can you add a brief description of the trade offs between the two choices? For instance, backwardeuler has a time step size constraint.

Also, for simplification, I would add the note about the default value here (and remove the following sentence), something like this:
`- ``laxwendroff`` is the semi-implicit scheme for E-update. (the default)`

Reply:
I read in Allen Taflove's Computational Electrodynamics book that the semi-implicit method is numerically stable for sigma from 0 to infinity. In the warp code, however, I read a comment that BE is more robust. I can change the default to BE, if that is more robust in your experience.

Reply:
In general for ODEs/PDEs when you look at semi-implicit-like (LW or Crank-Nicolson) approaches vs fully implicit (Backward Euler), all of these methods are formally stable for all size time steps.  However the semi-implicit approaches are more prone to developing oscillations (not an instability).  The flip side is that semi-implicit methods are 2nd order in time, whereas backward Euler is first order.  So I would agree that BE is more robust, but has a lower order of accuracy.